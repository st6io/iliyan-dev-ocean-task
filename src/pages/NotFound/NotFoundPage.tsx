import Layout from '@components/Layout';
import Section from '@components/Section';

const NotFountPage = () => (
  <Layout>
    <Section title="404" flex={1} textAlign="center">
      <Section.Text>Oops, page not found</Section.Text>
    </Section>
  </Layout>
);

export default NotFountPage;
