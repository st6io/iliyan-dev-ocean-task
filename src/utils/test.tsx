import { ReactElement, ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';

import { render as nativeRender } from '@testing-library/react';

const render = (ui: ReactElement, { path = '/', ...options } = {}) => {
  const RouterWrapper = ({ children }: { children: ReactNode }) => (
    <MemoryRouter initialEntries={[path]}>{children}</MemoryRouter>
  );

  return nativeRender(ui, {
    wrapper: RouterWrapper,
    ...options,
  });
};

export * from '@testing-library/react';
export { render };
