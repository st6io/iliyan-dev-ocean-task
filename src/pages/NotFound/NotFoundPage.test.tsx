import { render } from '@src/utils/test';

import NotFoundPage from './NotFoundPage';

describe('NotFoundPage', () => {
  it('should match snapshot', () => {
    const { container } = render(<NotFoundPage />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
