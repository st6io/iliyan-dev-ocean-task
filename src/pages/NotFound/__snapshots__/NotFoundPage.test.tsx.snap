// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should match snapshot 1`] = `
<div
  class="sc-cxabCf FKOKy"
>
  <header
    class="sc-efBctP eZxVPd"
  >
    <a
      href="/"
    >
      <img
        alt="home logo"
        class="sc-BeQoi"
        src="logo.png"
      />
    </a>
  </header>
  <div
    class="sc-cxabCf bsgFzP"
  >
    <section
      class="sc-bWXABl iFEscH"
    >
      <div
        class="sc-cxabCf sc-iJkHyd  bdVRyE"
      >
        404
      </div>
      <div
        class="sc-cxabCf sc-grVGCS  LTzaA"
      >
        Oops, page not found
      </div>
    </section>
  </div>
</div>
`;
